packages/@react-stately/color/test/useHexColorFieldState.test.js

```diff
@@ -0,0 +1,214 @@
+      red: 0,
+      green: 17,
+      blue: 35,
+      alpha: 1
+    });
+    expect(result.current.inputValue).toBe('#001123');
+  });
+
+  it('should not increment beyond the maximum value', function () {
+    let props = {defaultValue: '#abc', maxValue: '#abc'};
+    const {result} = renderHook(() => useHexColorFieldState(props));
+    act(() => result.current.increment());
+    expect(result.current.colorValue.value).toEqual({
+      red: 170,
+      green: 187,
+      blue: 204,
+      alpha: 1
+    });
+    expect(result.current.inputValue).toBe('#AABBCC');
+  });
+
+  it('should increment to max value', function () {
+    let props = {defaultValue: new Color('#aabbcc')};
+    const {result} = renderHook(() => useHexColorFieldState(props));
+    act(() => result.current.incrementToMax());
+    expect(result.current.colorValue.value).toEqual({
+      red: 255,
+      green: 255,
+      blue: 255,
+      alpha: 1
+    });
+    expect(result.current.inputValue).toBe('#FFFFFF');
+  });
+
+  it('should decrement', function () {
+    let props = {defaultValue: new Color('#012'), step: 4};
+    const {result} = renderHook(() => useHexColorFieldState(props));
+    act(() => result.current.decrement());
+    expect(result.current.colorValue.value).toEqual({
+      red: 0,
+      green: 17,
+      blue: 30,
+      alpha: 1
+    });
+    expect(result.current.inputValue).toBe('#00111E');
+  });
+
+  it('should not decrement beyond the minimum value', function () {
+    let props = {defaultValue: '#abc', minValue: '#abc'};
+    const {result} = renderHook(() => useHexColorFieldState(props));
+    act(() => result.current.decrement());
+    expect(result.current.colorValue.value).toEqual({
+      red: 170,
+      green: 187,
+      blue: 204,
+      alpha: 1
+    });
+    expect(result.current.inputValue).toBe('#AABBCC');
+  });
+
+  it('should decrement to min value', function () {
+    let props = {defaultValue: new Color('#aabbcc'), minValue: '#666'};
+    const {result} = renderHook(() => useHexColorFieldState(props));
+    act(() => result.current.decrementToMin());
+    expect(result.current.colorValue.value).toEqual({
+      red: 102,
+      green: 102,
+      blue: 102,
+      alpha: 1
+    });
+    expect(result.current.inputValue).toBe('#666666');
+  });
+
+  it('should revert to last valid value', function () {
+    let props = {defaultValue: '#abc', minValue: '#abc'};
+    const {result} = renderHook(() => useHexColorFieldState(props));
+    expect(result.current.inputValue).toBe('#AABBCC');
+    expect(result.current.colorValue.value).toEqual({
+      red: 170,
+      green: 187,
+      blue: 204,
+      alpha: 1
+    });
+
+    act(() => result.current.setInputValue('invalidColor'));
+    expect(result.current.inputValue).toBe('invalidColor');
+
+    act(() => result.current.commitInputValue());
+    expect(result.current.inputValue).toBe('#AABBCC');
+  });
+
+  it('should update colorValue (uncontrolled)', function () {
```

Comment:
I'm still trying to understand the difference between controlled and uncontrolled states, especially on how it'll work considering we always allow the user to type and `colorValue` gets updated (through `onChange`) when it gets valid inputs

Reply:
@snowystinger 
controlled: you manage
uncontrolled: component manages its own state